mlflow/statsmodels.py

```diff
@@ -473,61 +452,36 @@ def results_to_dict(results):
 
         return results_dict
 
-    def wrapper_fit(original_method):
-
-        """
-        External function to generate customized versions of fit with the proper value of the
-        original function (set externally in parameter). This enables a more accurate link
-        between the patched and the original function than using gorilla.get_original_attribute
-        in corner cases where `self` still refers to the subclass but the method we want to invoke
-        (in the context of the subclass) belongs to a superclass
-
-        :param original_method: the original function object that will be replaced by this function
-        :return: the new fit function, from which we will be doing a call to the original fit
-                 method at some point
-        """
-
-        def fit(self, *args, **kwargs):
+    def wrapper_fit(original, self, *args, **kwargs):
```

Comment:
I don't think there's a need for the patch implementation to call the `original_method` of a superclass from a subclass. As discussed in https://github.com/mlflow/mlflow/pull/3859/files#r544936250, `safe_patch` will handle parent/child patch ordering by way of `mlflow.utils.gorilla` and will deliver the correct `original` function as the first parameter to the patch implementation (`original`).

@harupy Please let me know if I'm missing anything here.